src/gsy_e/gsy_e_core/sim_results/endpoint_buffer.py

```diff
@@ -195,14 +196,35 @@ def _create_results_handler(should_export_plots):
     def create_hierarchy_stats(self, area: "AreaBase"):
         """Calculate hierarchy related statistics."""
         hierarchy_self_consumption_list = {}
-        self._calc_hierarchy_stats(area, 0, hierarchy_self_consumption_list)
+        hierarchy_self_consumption = {}
+        self._calc_hierarchy_self_consumption(area, 0, hierarchy_self_consumption_list)
         for level, consumption_list in hierarchy_self_consumption_list.items():
             if len(consumption_list):
-                self.hierarchy_self_consumption[level] = sum(consumption_list) / len(
-                    consumption_list
-                )
+                hierarchy_self_consumption[level] = sum(consumption_list) / len(consumption_list)
+        if (
+            len(hierarchy_self_consumption) > 0
+            and hierarchy_self_consumption[next(iter(hierarchy_self_consumption))] == 0
+        ):
+            # early return in case there is no self consumpton at all (in order to not devide by 0)
+            return
 
-    def _calc_hierarchy_stats(self, area: "AreaBase", level: int, results: Dict):
+        for level, self_consumption in hierarchy_self_consumption.items():
+            if level + 1 not in hierarchy_self_consumption:
```

Comment:
Not that big of an issue, since it probably makes the solution simpler, however I think that `hierarchy_self_consumption` could be a list and not a dict, since the index / position in the list could be used as the level number. Regardless if you believe it is more explicit that way leave it as is.

Reply:
I do not mind either way, so I changed it to be a list: https://github.com/gridsingularity/gsy-e/pull/1800/commits/87ff961819ddf20482f0f1f6fd929f622c390d18